homeassistant/components/netgear_lte/__init__.py

```diff
@@ -137,62 +148,116 @@ def get_modem_data(self, config):
-            hass, cookie_jar=aiohttp.CookieJar(unsafe=True)
-        )
-        hass.data[DOMAIN] = LTEData(websession)
-
-        await async_setup_services(hass)
-
-    netgear_lte_config = config[DOMAIN]
-
-    # Set up each modem
-    tasks = [
-        hass.async_create_task(_setup_lte(hass, lte_conf))
-        for lte_conf in netgear_lte_config
-    ]
-    await asyncio.wait(tasks)
-
-    # Load platforms for each modem
-    for lte_conf in netgear_lte_config:
-        # Notify
-        for notify_conf in lte_conf[CONF_NOTIFY]:
-            discovery_info = {
-                CONF_HOST: lte_conf[CONF_HOST],
-                CONF_NAME: notify_conf.get(CONF_NAME),
-                CONF_NOTIFY: notify_conf,
-            }
+    hass.data[DATA_HASS_CONFIG] = config
+
+    if lte_config := config.get(DOMAIN):
+        for entry in lte_config:
             hass.async_create_task(
-                discovery.async_load_platform(
-                    hass, Platform.NOTIFY, DOMAIN, discovery_info, config
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
                 )
             )
-
-        # Sensor
-        sensor_conf = lte_conf[CONF_SENSOR]
-        discovery_info = {CONF_HOST: lte_conf[CONF_HOST], CONF_SENSOR: sensor_conf}
-        hass.async_create_task(
-            discovery.async_load_platform(
-                hass, Platform.SENSOR, DOMAIN, discovery_info, config
-            )
+        async_create_issue(
+            hass,
+            DOMAIN,
+            "deprecated_yaml",
+            breaks_in_ha_version="2023.9.0",
+            is_fixable=False,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
         )
 
-        # Binary Sensor
-        binary_sensor_conf = lte_conf[CONF_BINARY_SENSOR]
-        discovery_info = {
-            CONF_HOST: lte_conf[CONF_HOST],
-            CONF_BINARY_SENSOR: binary_sensor_conf,
-        }
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Netgear LTE from a config entry."""
+    host = entry.data[CONF_HOST]
+    password = entry.data[CONF_PASSWORD]
+
+    websession = async_create_clientsession(hass, cookie_jar=CookieJar(unsafe=True))
+
+    hass.data[DOMAIN] = LTEData(websession)
+
+    modem = Modem(
+        hostname=host,
+        password=password,
+        websession=websession,
+    )
+    try:
+        await modem.login()
+    except Error as ex:
+        raise ConfigEntryNotReady("Cannot connect/authenticate") from ex
+
+    # Set up modem
+    await _setup_lte(hass, entry.data)
+
+    await async_setup_services(hass)
+
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            Platform.NOTIFY,
+            DOMAIN,
+            {CONF_HOST: entry.data[CONF_HOST], CONF_NAME: entry.title},
+            hass.data[DATA_HASS_CONFIG],
+        )
+    )
```

Comment:
Discovery of notify happens three times, please add comments explaining the differences.